src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -180,8 +205,30 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
 
                         using (X509Certificate2 leafCert = new X509Certificate2(leafCertPtr))
```

Comment:
@davidsh, this code here:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs#L260-L292
It creates the X509Certificate2 but never disposes of it, leaving it for finalization.  If, however, it is disposed, then this test code fails:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http.WinHttpHandler/tests/FunctionalTests/ServerCertificateTest.cs#L122
as it's called after the callback has already returned yet expects the certificate to not have been disposed.